ranger2/aws/ranger_asg/main.tf

```diff
@@ -0,0 +1,579 @@
+  depends_on                  = [aws_db_instance.ranger_mysql]
+  tags = { 
+    Name = var.ranger_admin_ami_name
+  }
+
+  provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "ec2-user"
+      host        = aws_instance.tf_ranger_base_inst.public_dns
+      private_key = file("${path.module}/pskey.pem")
+    }
+
+    inline = [
+      "sleep 300;while [ ! -f /tmp/setup-done ]; do echo 'Waiting for ranger setup script to complete'; sleep 20; done;"
+    ]
+  }
+
+  user_data = data.template_file.ranger_base_ami_tmpl.rendered
+}
+
+output "ranger_base_inst_id" {
+  value = aws_instance.tf_ranger_base_inst.id
+}
+
+#Create Ranger AMI 
+resource "aws_ami_from_instance" "tf_ranger_ami" {
+  name               = var.ranger_admin_ami_name
+  source_instance_id = aws_instance.tf_ranger_base_inst.id
+  tags = { 
+    Name = var.ranger_admin_ami_name
+  }
+}
+
+output "ranger_ami_id" {
+  value = aws_ami_from_instance.tf_ranger_ami.id
+}
+
+#Create Ranger Launch Template
+resource "aws_launch_template" "tf_ranger_lt" {
+  name          = var.ranger_lt_name
+  image_id      = aws_ami_from_instance.tf_ranger_ami.id
+  instance_type = var.ranger_instance_type
+  key_name      = var.key_name
+  depends_on    = [aws_ami_from_instance.tf_ranger_ami]
+
+  network_interfaces {
+    associate_public_ip_address = true
+    delete_on_termination = true
+    device_index = 0
+    subnet_id = var.public_subnets[1]
+    security_groups = [aws_security_group.ec2_sg.id]
+  }
+
+  tag_specifications {
+    resource_type = "instance"
+    tags = {
+      Name = var.ranger_admin_name
+    }
+  }
+
+  user_data = filebase64("${path.module}/ranger_start.sh")
+}
+
+resource "aws_autoscaling_group" "tf_ranger_asg" {
+    name                      = var.ranger_asg_name
+    #count                     = var.ranger_inst_cnt
+    min_size                  = var.def_inst_cnt
+    max_size                  = var.def_inst_cnt
+    desired_capacity          = var.def_inst_cnt
+    #vpc_zone_identifier       = [var.public_subnets[0],var.public_subnets[1]]
+    vpc_zone_identifier       = var.public_subnets
+    health_check_grace_period = 10 
+    health_check_type         = "EC2"
+    default_cooldown          = 10
+    wait_for_elb_capacity     = 0
+    wait_for_capacity_timeout = "10m"
+    target_group_arns         = [aws_lb_target_group.ranger_alb_tg.arn]
+    depends_on                = [aws_launch_template.tf_ranger_lt]
+    launch_template {
+      id      = aws_launch_template.tf_ranger_lt.id
+      version = "$Latest"
+    }
+    tag {
+      key = "Name"
+      value = var.ranger_admin_name
+      propagate_at_launch = true
+    }
+}
+
+resource "null_resource" "create-default-ranger-policy" {
+    depends_on = [aws_autoscaling_group.tf_ranger_asg]
+    provisioner "local-exec" {
+      command = "sleep 240;python ranger_policy.py ${aws_lb.ranger_alb.dns_name} ${var.def_loc} ${var.service_name} ${var.ranger_alb_port}"
```

Comment:
how will we catch failures in the py file execution?

Reply:
Exceptions will be displayed on the console